### PR TITLE
added get access token

### PR DESCRIPTION
**Pull request by AlonBucks**

### CODE REVIEW

File: intezeranalyze_connector.py [hunk 1]
```diff
@@ -33,6 +35,8 @@ def __init__(self):
         # modify this as you deem fit.
         self._base_url = None
```
Comment (davidt99) [inline intezeranalyze_connector.py:36]: make sure the base url contains /api like https://analyze.intezer.com/api/ or add it to the rest of the routes

File: intezeranalyze_connector.py [hunk 2]
```diff
@@ -362,8 +371,7 @@ def _handle_file_reputation(self, param):
 
         # Parameters to send to Intezer
         params = {
-            'sha256': sha_hash,
-            'api_key': api_key
+            'sha256': sha_hash
```
Comment (davidt99) [inline intezeranalyze_connector.py]: change to `hash`

File: intezeranalyze_connector.py [hunk 3]
```diff
@@ -124,6 +128,19 @@ def _make_rest_call(self, endpoint, action_result, headers=None, params=None, da
         # Create a URL to connect to
         url = self._base_url + endpoint
 
+        if endpoint != 'is-available' and not self._jwt_token:
```
Comment (davidt99) [inline intezeranalyze_connector.py:130]: there are still places the api is used as a parameter (line 364, 393)

File: intezeranalyze_connector.py [hunk 4]
```diff
@@ -124,6 +128,19 @@ def _make_rest_call(self, endpoint, action_result, headers=None, params=None, da
...
+            response = requests.post(self._base_url + 'v2-0/get-access-token', json={'api_key': config.get('apiKey')})
+            if response.status_code != HTTPStatus.OK:
+                return RetVal(action_result.set_status(phantom.APP_ERROR,
+                                                       "Error Connecting to server. Details: {0}".format(str(response.text))),
+                              resp_json)
+            self._jwt_token = response.json()['result']
```
Comment (davidt99) [inline intezeranalyze_connector.py]: i'm not sure how this is implemented in phantom, but jwt can expire if this class is up for more than 10 hours. catch 401 responses and retry one more time after get access token

File: intezeranalyze_connector.py [hunk 5]
```diff
@@ -124,6 +128,19 @@ def _make_rest_call(self, endpoint, action_result, headers=None, params=None, da
         # Create a URL to connect to
         url = self._base_url + endpoint
 
+        if endpoint != 'is-available' and not self._jwt_token:
+            response = requests.post(self._base_url + 'v2-0/get-access-token', json={'api_key': config.get('apiKey')})
```
Comment (davidt99) [inline intezeranalyze_connector.py]: i thins we can add the v2-0 to the base url

**Review by davidt99 — CHANGES_REQUESTED** [review]

File: intezeranalyze_connector.py (hunk 5 above)
Comment (AlonBucks) [reply intezeranalyze_connector.py]: we can't because of the is-available endpoint which is without v2-0

File: intezeranalyze_connector.py (hunk 1 above)
Comment (AlonBucks) [reply intezeranalyze_connector.py:36]: it is contain /api, it's located on phintezeranalyze.json

File: intezeranalyze_connector.py [hunk 6]
```diff
@@ -110,43 +114,59 @@ def _process_response(self, r, action_result):
...
 
         config = self.get_config()
 
-        resp_json = None
-
         try:
             request_func = getattr(requests, method)
         except AttributeError:
-            return RetVal(action_result.set_status(phantom.APP_ERROR, "Invalid method: {0}".format(method)), resp_json)
+            return RetVal(action_result.set_status(phantom.APP_ERROR, "Invalid method: {0}".format(method)), None)
```
Comment (davidt99) [inline intezeranalyze_connector.py]: no reason to format none

File: intezeranalyze_connector.py [hunk 7]
```diff
@@ -203,24 +217,21 @@ def _handle_detonate_file(self, param):
...
+
         # Check file type is correct
         if file_check:
-            file_list = {'file': open(str_vault_dict, 'rb')}
+            file = {'file': open(str_vault_dict, 'rb')}
             self.save_progress("File has been found location is: {}".format(str_vault_dict))
 
         # Make connection to the Intezer Analyze endpoint
-        ret_val, response = self._make_rest_call(endpoint, action_result, method="post", data=params, files=file_list)
+        ret_val, response = self._make_rest_call(endpoint, action_result, method="post", params=file)
```
Comment (davidt99) [inline intezeranalyze_connector.py:238]: if file_check is false, we do a call with no file...

File: intezeranalyze_connector.py [hunk 8]
```diff
@@ -235,10 +246,10 @@ def _handle_detonate_file(self, param):
             analysis_id = response['analysis_id']
```
Comment (davidt99) [inline intezeranalyze_connector.py]: we no longer have it in the response, use the result_url instead

File: intezeranalyze_connector.py [hunk 9]
```diff
@@ -288,25 +299,14 @@ def _handle_get_report(self, param):
...
         report_id = param['id']
 
         # Issue request to Intezer Analyze
-        endpoint = 'v2-0/analyses/{}'.format(report_id)
-
-        # Parameters to send to Intezer
-        params = {
-            'api_key': api_key
-        }
+        endpoint = 'analyses/{}'.format(report_id)
```
Comment (davidt99) [inline intezeranalyze_connector.py:308]: same here

File: intezeranalyze_connector.py [hunk 10]
```diff
@@ -320,9 +320,6 @@ def _handle_get_report(self, param):
             # Pass Analysis ID to Intezer Analyze
             analysis_id = response['analysis_id']
```
Comment (davidt99) [inline intezeranalyze_connector.py]: same

File: intezeranalyze_connector.py [hunk 11]
```diff
@@ -381,13 +371,11 @@ def _handle_file_reputation(self, param):
...
```
Comment (davidt99) [inline intezeranalyze_connector.py]: same

**Review by davidt99 — CHANGES_REQUESTED** [review]

**Review by davidt99 — APPROVED** [review]

**Pull request merged**